### ISSUE TITLE
nest-simulator compile failing

### ISSUE THREAD

**Issue by dnao** [nest/nest-simulator#2431]
Hi,
I am trying to compile the latest version of nest simulator on a fresh conda environment and would appreciate some assistance as it is failing.

I am using the following packages in conda:

`conda create --name condapythonfornestml -c conda-forge ipython cxx-compiler pyqt wxpython numpy pandas matplotlib h5py jsonschema pytest boost`

When I run cmake it failes at 99%:
```
[ 99%] Linking CXX executable run_all_cpptests
/import/headnode1/dnao/suphys/anaconda/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: CMakeFiles/run_all_cpptests.dir/run_all.cpp.o: undefined reference to symbol 'clock_gettime@@GLIBC_2.2.5'
/import/headnode1/dnao/suphys/anaconda/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: /import/headnode1/dnao/suphys/anaconda/bin/../x86_64-conda_cos6-linux-gnu/sysroot/lib/librt.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [testsuite/cpptests/run_all_cpptests] Error 1
make[1]: *** [testsuite/cpptests/CMakeFiles/run_all_cpptests.dir/all] Error 2
make: *** [all] Error 2

```

I have attached the logfile with the configuration: [logfile.txt](https://github.com/nest/nest-simulator/files/9128170/logfile.txt)

**Comment by terhorstd** [comment]
Can you check if you have all required packages in stalled as mentioned in the Conda `environment.yml`? Check the Conda tab in the [install docs](https://nest-simulator.readthedocs.io/en/v3.3/installation/index.html). It looks like an incompatible `glibc` somewhere in your environment.

**Comment by dnao** [comment]
Thank you for the assistance @terhorstd . I have tried to compile again using a cluster node running RHEL7.9. This time it fails at 47%. I have attached the make and cmake output.

Pre-installed software includes gsl-2.7, cmake 3.24.0 and openMPI-1.10.3-gnu

I have built a fresh conda environment with the following packages:
conda create --name pythonfornestml -c conda-forge boost readline ipython pyqt wxpython cython


[make_output.txt](https://github.com/nest/nest-simulator/files/9260227/make_output.txt)
[cmake_output.txt](https://github.com/nest/nest-simulator/files/9260220/cmake_output.txt)

I note your caution regarding using conda on a cluster environment. I will try to compile the dependencies from source instead to see if that fixes the issue.
